docs/src/idiomatic/007if-then-else.md

```diff
@@ -0,0 +1,94 @@
+
+State formulas are formulas that don't contain any action operator (e.g. primed variables, `UNCHANGED`). Using `IF-THEN-ELSE` on this type of formula can make it easier to read in some situations, and don't come with any disadvantage. This example state formula uses `IF-THEN-ELSE` to return a boolean value: 
+
+```tla
+ValidIdentity(person) == IF Nationalized(person) THEN ValidId(person) ELSE ValidPassport(person)
+```
+
+Although it could be rewritten with boolean operators, it doesn't read as nicely:
+
+```tla
+ValidIdentity(person) == \/ /\ Nationalized(person)
+                            /\ ValidId(person)
+                         \/ /\ !Nationalized(person)
+                            /\ ValidPassport(person)
+```
+
+## When (and how) *not* to use `IF-THEN-ELSE`
+
+Mixing `IF-THEN-ELSE` expressions with action operators introduces unnecessary branching to definitions that could be self-contained and look more like a transition definition. 
+
+```tla
+Withdraw(amount) == IF balance >= amount 
+                     THEN /\ balance' = balance - amount
+                          /\ response' = "SUCCESS"
+                     ELSE /\ UNCHANGED<balance>
+                          /\ response' = "FAILURE"
+```
+
+We could separate the two branches into their own actions with clarifying names and explicit conditions, and use a disjunction between the two actions instead of the `IF-THEN-ELSE` block:
+
+```tla
+WithdrawSuccess(amount) == /\ balance >= amount 
+                           /\ balance' = balance - amount
+                           /\ response' = "SUCCESS"
+                            
+WithdrawFailure(amount) == /\ balance < amount 
+                           /\ response' = "FAILURE"
+                           /\ UNCHANGED<balance> 
+                            
+Withdraw(amount) == WithdrawSuccess(amount) \/ WithdrawFailure(amount)
+```
+
+## Advantages
+
+- Each action declares fewer transitions, so it's easier to reason about it
+- A disjunction of actions is closer to a union of transition sets than an `IF-THEN-ELSE` expression
+- Nested `IF-THEN-ELSE` expressions are an extrapolation of these problems and can over-constrain some branches if not done carefully. Using different actions defining its conditions explicitly leaves less room for implicit wrong constraints that an `ELSE` branch allows. See the example below.
+
+Assuming `C1()` is a condition for `A1()` and `C2()` is a condition for `A2()`:
+
+```tla
+Next == IF C1() 
+          THEN A1()
+          ELSE 
+            IF C2() 
+              THEN A2()
+              ELSE A3()
+```
+
+What if `C1() /\ C2()` is true? In this case, only `A1()` would be enabled, which is incorrect. 
+
+```tla
+Next == \/ /\ C1()
+           /\ A1()
+        \/ /\ C2()
+           /\ A2()
+        \/ A3()
+           
+```
+
+This second definition can allow more behaviors than the first one (depending on whether `C1()` and `C2()` overlap), and therefore is safer in most cases.
```

Comment:
This is a bit disingenuous, IMO. If you _wanted_ to define the behavior written in ITE, your disjunction would be _less_ readable, not more, and more prone to errors.
>This second definition can allow more behaviors ... and therefore is safer in most cases.

I disagree, the assumption that allowing more behaviors is safer is questionable.

Reply:
> If you wanted to define the behavior written in ITE
This is a idiom, the aim is to help people decide what they want when they are not sure. I wrote this assuming a lot of people use ITE just because they are way more used to programming languages than logic. My point in this example is: if you have a bunch of independent "when the state is this, do that" , its better to use a disjunction, but most people will use ITE because that's how you do it in programming languages. 

As for allowing more behaviors being safer: yeah, that's kinda pretentious, I'll get rid of it.

Reply:
I think Gabriela's point here is that the author of such a spec would not have thought about the case of `A1` and `A2` being simultaneously enabled. A disjunction makes it clear. I agree that it is often the case that people wants to specify a concurrent system via `IF-THEN-ELSE`, which is a purely sequential operator.